lib/cylc/network/server.py

```diff
@@ -258,65 +293,140 @@ def __init__(self, schd):
 
-    def get_priv_level(self, user):
+    def _get_priv_level(self, user):
         """Return the privilege level for the given user for this suite."""
         if user == getpass.getuser():
             return Priv.CONTROL
         if self.public_priv is None:
             # cannot do this on initialisation as the suite configuration has
             # not yet been parsed
-            self.public_priv = self.get_public_priv()
+            self.public_priv = self._get_public_priv()
         return self.public_priv
 
+    @authorise(Priv.IDENTITY)
+    @ZMQServer.expose
+    def api(self, endpoint=None):
+        """Return information about this API.
+
+        Returns a list of callable endpoints.
+
+        Args:
+            endpoint (str, optional):
+                If specified the documentation for the endpoint
+                will be returned instead.
+
+        Returns:
+            list/str: List of endpoints or string documentation of the
+            requested endpoint.
+
+        """
+        if not endpoint:
+            return [
+                method for method in dir(self)
+                if getattr(getattr(self, method), 'exposed', False)
+            ]
+
+        try:
+            method = getattr(self, endpoint)
+        except AttributeError:
+            return 'No method by name "%s"' % endpoint
+        if method.exposed:
+            head, tail = method.__doc__.split('\n', 1)
+            tail = dedent(tail)
+            return '%s\n%s' % (head, tail)
+        return 'No method by name "%s"' % endpoint
+
     @authorise(Priv.CONTROL)
     @ZMQServer.expose
     def clear_broadcast(
             self, point_strings=None, namespaces=None, cancel_settings=None):
         """Clear settings globally, or for listed namespaces and/or points.
 
-        Return a tuple (modified_settings, bad_options), where:
-        * modified_settings is similar to the return value of the "put" method,
-          but for removed settings.
-        * bad_options is a dict in the form:
-              {"point_strings": ["20020202", ..."], ...}
-          The dict is only populated if there are options not associated with
-          previous broadcasts. The keys can be:
-          * point_strings: a list of bad point strings.
-          * namespaces: a list of bad namespaces.
-          * cancel: a list of tuples. Each tuple contains the keys of a bad
-            setting.
+        Args:
+            point_strings (list, optional):
+                List of point strings for this operation to apply to or
+                ``None`` to apply to all cycle points.
+            namespaces (list, optional):
+                List of namespace string (task / family names) for this
+                operation to apply to or ``None`` to apply to all namespaces.
+            cancel_settings (list, optional):
+                List of broadcast keys to cancel.
+
+        Returns:
+            tuple: (modified_settings, bad_options)
```

Comment:
Ditto above about limitation on return types. Possibly using PEP 484 and some different annotation would allow IDE and generated documentation to match. Right now IDE doesn't understand the tuples or multiple return types. At least the generated documentation is clear (thanks for that!) :)